### PR TITLE
Map preview

### PR DESCRIPTION
**Pull request by ellie9393**
In this PR, I added a map preview box on the home page. It gets updated based on the user input in the search input. There is a button that is supposed to be on the bottom right of the box, which directs the user to the map page. But it doesn't appear on the box!! and I wasn't able to fix it, Theoretically, it should work, but I can't see what's the issue. 
So this PR is not completed.
Screenshot 2023-11-29 at 9 42 05 AM

### CODE REVIEW

File: src/components/MapPreview/index.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      >
+      <TileLayer {...tileLayer} />
+      </MapContainer>
+
+      <Link to="/Maps" style={{ textDecoration: "none" }}>
+        <Button
+          variant="contained"
+          color="primary"
+          sx={{
+            position: "absolute",
```
Comment (mnxn) [inline src/components/MapPreview/index.tsx:41]: The `absolute` position makes it relative to the entire page, which is why the button is so far away from the map.

You should be able to fix it by changing it to a `relative` position. Then to get it to show up on top of the map, you need to set a `zIndex` property to a high value.

File: src/components/MapPreview/index.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+interface MapPreviewProps {
+  center: { lat: number; lng: number };
+}
+
+const MapPreview = ({ center}: MapPreviewProps) => {
+  return (
+    <Box
+      sx={{
+        height: "200px",
```
Comment (mnxn) [inline src/components/MapPreview/index.tsx:16]: It's a little odd that the map is so much shorter than the current location component.
I think you should be able to set this to a `"100%"` height for it to fill the full height.

File: src/components/MapPreview/index.tsx (hunk 1 above)
Comment (ellie9393) [reply src/components/MapPreview/index.tsx]: I had tried the relative position, but when I'd resized the page, the button would move. 
I wrapped it in another box container, and it fixed my issue. 

Screenshot 2023-11-29 at 10 24 03 AM

File: src/components/MapPreview/index.tsx (hunk 2 above)
Comment (ellie9393) [reply src/components/MapPreview/index.tsx]: yes, that's right. I changed it.

**Comment by mnxn** [comment]
Do you think you could move the button to the top right or bottom left so it does not have have the attribution drawn over it?

You could also use an icon to save space: https://mui.com/material-ui/material-icons/?query=expand&selected=OpenInFull

**Review by mnxn — APPROVED** [review]
Forgot to approve, but you can merge now too.

File: src/components/MapPreview/index.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+interface MapPreviewProps {
+  center: { lat: number; lng: number };
+}
+
+const MapPreview = ({ center}: MapPreviewProps) => {
+  return (
+    <Box
+      sx={{
+        position: "relative",
+        height: "100%",
```
[the remark this answers is not in the document]
Comment (ellie9393) [reply src/components/MapPreview/index.tsx:19]: It works for me, too.

**Pull request merged**